cmd/system-probe/modules/network_tracer.go

```diff
@@ -80,11 +99,89 @@ type networkTracer struct {
 	restartTimer *time.Timer
 }
 
+func startRPCServer(tracer networkTracer) error {
+	socketFile := "/tmp/my_grpc.sock"
+
+	// Remove existing socket file if it exists
+	os.Remove(socketFile)
+
+	listener, err := net.Listen("unix", socketFile)
+	if err != nil {
+		return err
+	}
+
+	server := grpc.NewServer()
+	test2.RegisterSystemProbeServer(server, &tracer)
+
+	useGRPCServer := dd_config.SystemProbe.GetBool("service_monitoring_config.use_grpc")
```

Comment:
if the value is false, we should reach to `startRPCServer` from the first place

Reply:
BTW, you should export it to the configuration object, and fetch it from it

Reply:
> if the value is false, we should reach to `startRPCServer` from the first place

I think I fixed the issue if I understood you comment correctly

Reply:
you've made it partially, let's discuss by heart